ci: emit sg lint honeycomb timings in CI (temporary)

Strum355 opened the pull request:
Temporarily emit telemetry on sg-lint steps to honeycomb to determine timings


## Test plan

https://ui.honeycomb.io/sourcegraph/datasets/sg-lint-buildkite/result/FeP9mXgMxzB

jhchabran commented:
keep in mind that I added a mutex to sync the two pnpm tasks, so the numbers might look weird for those :)

Strum355 commented:
> keep in mind that I added a mutex to sync the two pnpm tasks, so the numbers might look weird for those :)

I added some specific numbers to be able to check that too https://github.com/sourcegraph/sourcegraph/pull/59589/files#diff-bbba3d02dc06339eaa101e0bc633c74676398dd9652e94fea0aab25a64757892R157
